### ISSUE TITLE
Window size on Linux Fedora 41 - Dell XPS 13 is minuscule -> not adapting ot display

### ISSUE THREAD

**Issue by giulio-bellizia** [processing/processing4#857]
## Description

After installing Processing 4.3.1 on a Dell XPS 13 9340 with Linux Fedora 41, when I open the application the window is very small with almost impossible font to read
## Expected Behavior

I'd expect the window to size according to the display in use (a laptop, in this case)
## Current Behavior

Very small window when opening the application. The application seems not to capture the display resolution/settings correctly
## Steps to Reproduce

1. install the application from the desktop software store
2. open it
3. the windows it minuscule

## Your Environment
 
 

* Processing version: 4.3.1
* Operating System and OS version: Linux Fedora 41
* Other information: Dell XPS 13 9340 Laptop

**Comment by SableRaf** [comment]
Hi @giulio-bellizia,

Your report is much appreciated. I have transferred your issue to the [processing/processing4](https://github.com/processing/processing4/) repository. For issues with Processing 4, please use: https://github.com/processing/processing4/issues

Thanks!

**Comment by SableRaf** [comment]
@giulio-bellizia Would you mind sharing a screenshot for reference?

**Comment by giulio-bellizia** [comment]
![Pasted image](https://github.com/user-attachments/assets/888d1414-5283-4e0a-988f-2acc33e6b991)

I attach here a screenshot

**Comment by hx2A** [comment]
@giulio-bellizia , what window manager are you using? Gnome?

I am upgrading my computer from Fedora 40 to 41 this weekend. I'll have a look at this after the upgrade.

**Comment by hx2A** [comment]
I seem to be able to reproduce this problem on Fedora 41 when I scale the display in my Display Configuration. Should the PDE read the pixel density when it opens a window? How does that sort of thing work on macOS machines with retina displays?

**Comment by Stefterv** [comment]
I did some looking into this, adding `System.setProperty("sun.java2d.uiScale","2")` before any swing is instantiated will fix it, I just have not found a reliable method to find out what the current scaling on Linux is. If anyone knows we could add it as a fix

**Comment by hx2A** [comment]
Is this something I can help you with? I have a Linux machine with Fedora 41. 

Right now I see `System.getProperty("sun.java2d.uiScale")` returns null.

**Comment by Stefterv** [comment]
Help is always appreciated @hx2A 😄 

Do you have enough info to start a search?

**Comment by hx2A** [comment]
No, I have zero information. I do have a Linux machine with Fedora 41, so I can answer questions about what is happening on this machine.

**Comment by Stefterv** [comment]
Ah sorry, I've added a pull request with something that temporarily fixes is but has some a regression for platforms not running at 2x scale. I'm basically looking for a method on your machine that will give us the indication that it is running at 2x scale so we can activate this method.

Feel free to try the build here https://github.com/processing/processing4/actions/runs/12977296278?pr=934

**Comment by mingness** [comment]
Hey @Stefterv, I currently have this problem using Pop OS 24.04 LTS. Happy to work on solving this problem.

**Comment by mingness** [comment]
I suggest a two part solution. First, a generic zoom functionality, which would work towards improved accessibility. Second, a closer look at how scale is stored for different linux distros. Does that sound ok? If so, I'll work first on a zoom functionality, that could be inserted into the Edit menu, after the font size adjustments. If you have any design thoughts, or tech tips on how to do this, please share.

Is there a how-to on how to build the application locally?

**Comment by SableRaf** [comment]
> I'll work first on a zoom functionality, that could be inserted into the Edit menu

Doesn't the PDE already have a UI scale feature?

Build instructions can be found here:
https://github.com/processing/processing4/blob/main/BUILD.md

**Comment by mingness** [comment]
> Doesn't the PDE already have a UI scale feature?

Ah, found it in the preferences. Thanks. The one thing that doesn't scale is the preference window itself. and the code text needs to be set separately. Perhaps this interaction  could be smoothed.
